das/store.go

```diff
@@ -0,0 +1,108 @@
+package das
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+)
+
+var (
+	storePrefix   = datastore.NewKey("das")
+	checkpointKey = datastore.NewKey("checkpoint")
+)
+
+// The checkpoint store stores/loads the DASer's checkpoint to/from
+// disk using the checkpointKey. The checkpoint is stored as a struct
+// representation of the latest successfully DASed state.
+type store struct {
+	datastore.Datastore
+}
+
+// backgroundStore periodically saves current sampling state in case of DASer force quit before
+// being able to store state on exit
+type backgroundStore struct {
+	done
+}
```

Comment:
This already has an open issue #1049

Reply:
But very very strongly agree.

Also, now the store is being passed into backgroundStore.run instead of being part of the struct? I don't understand this change